packages/blockchain-wallet-v4-frontend/src/scenes/Nfts/Asset/Asset.tsx

```diff
@@ -621,28 +621,57 @@ const NftAsset: React.FC<Props> = ({
-                            asset_contract_address: contract,
-                            step: NftOrderStepEnum.MARK_FOR_SALE,
-                            token_id: id,
-                            walletUserIsAssetOwnerHack: true
-                          })
-                          analyticsActions.trackEvent({
-                            key: Analytics.NFT_MARK_FOR_SALE,
-                            properties: {
-                              collection: collectionName,
-                              collection_id: id
-                            }
-                          })
-                        }}
-                      >
-                        <FormattedMessage id='copy.mark_for_sale' defaultMessage='Mark for Sale' />
-                      </Button>
+                      {!lowest_order ? (
+                        <Button
+                          data-e2e='openNftFlow'
+                          nature='primary'
+                          jumbo
+                          onClick={() => {
+                            nftsActions.nftOrderFlowOpen({
+                              asset_contract_address: contract,
+                              step: NftOrderStepEnum.MARK_FOR_SALE,
+                              token_id: id,
+                              walletUserIsAssetOwnerHack: true
+                            })
+                            analyticsActions.trackEvent({
+                              key: Analytics.NFT_MARK_FOR_SALE,
+                              properties: {
+                                collection: collectionName,
+                                collection_id: id
+                              }
+                            })
+                          }}
+                        >
+                          <FormattedMessage
+                            id='copy.mark_for_sale'
+                            defaultMessage='Mark for Sale'
+                          />
+                        </Button>
+                      ) : (
+                        <Button
+                          data-e2e='openNftFlow'
+                          nature='primary'
+                          jumbo
+                          onClick={() =>
+                            nftsActions.nftOrderFlowOpen({
+                              asset_contract_address: contract,
+                              offer: undefined,
+                              order: lowest_order,
+                              step: NftOrderStepEnum.CANCEL_LISTING,
+                              token_id: id,
+                              walletUserIsAssetOwnerHack: true
+                            })
+                          }
+                        >
+                          <Text color={colors.blue600} size='14px' weight={500}>
+                            <FormattedMessage
+                              id='buttons.cancel_offer'
+                              defaultMessage='Cancel Listing'
```

Comment:
you need to make sure the id's and default messages always match everywhere in the app.  this is going to cause a conflict